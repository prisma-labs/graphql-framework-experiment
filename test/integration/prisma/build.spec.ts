import { createWorkspace } from '../../__helpers'

const ws = createWorkspace({
  name: 'prisma-build',
})

it('can build a prisma framework project', () => {
  ws.fs.write(
    'schema.prisma',
    `	
      datasource db {	
        provider = "sqlite"	
        url      = "file:dev.db"	
      }	
      generator photon {	
        provider = "photonjs"	
      }	
      model User {	
        id   Int    @id	
        name String	
      }	
    `
  )

  ws.fs.write(
    'schema.ts',
    `
      objectType({
        name: 'User',
        definition(t) {
          t.model.id()
          t.model.name()
        }
      })
    `
  )

  const result = ws.run('yarn -s pumpkins build')
  expect(result).toMatchSnapshot()
  expect(ws.fs.inspectTree('dist')).toMatchSnapshot()
})
