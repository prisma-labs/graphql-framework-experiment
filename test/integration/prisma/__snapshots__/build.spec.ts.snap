// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can build a prisma framework project 1`] = `
Object {
  "status": 0,
  "stderr": "",
  "stdout": "ðŸŽƒ  Running Prisma generators ...
ðŸŽƒ  Generating Nexus artifacts ...
ðŸŽƒ  Compiling ...
ðŸŽƒ  Pumpkins server successfully compiled!
",
}
`;

exports[`can build a prisma framework project 2`] = `
Object {
  "children": Array [
    Object {
      "name": "schema.js",
      "size": 239,
      "type": "file",
    },
    Object {
      "name": "start.js",
      "size": 507,
      "type": "file",
    },
  ],
  "name": ".build",
  "size": 746,
  "type": "dir",
}
`;
